src/AbstractDifferentiation.jl

```diff
@@ -224,63 +247,200 @@ struct LazyDerivative{B, F, X}
+
 function Base.:*(ys, d::LazyJacobian)
     if ys isa Tuple
         ya = adjoint.(ys)
     else
         ya = adjoint(ys)
     end
-    return pullback_function(d.ab, d.f, d.xs...)(ya)
+    if d.xs isa Tuple
+        return pullback_function(d.backend, d.f, d.xs...)(ya)
+    else
+        return pullback_function(d.backend, d.f, d.xs)(ya)
+    end
+end
+
+function Base.:*(d::LazyJacobian, ys::Number)
+    if d.xs isa Tuple
+        return jacobian(d.backend, d.f, d.xs...) .* ys
+    else
+        return jacobian(d.backend, d.f, d.xs) .* ys
+    end
 end
 
+function Base.:*(ys::Number, d::LazyJacobian)
+    if d.xs isa Tuple
+        return jacobian(d.backend, d.f, d.xs...) .* ys
+    else
+        return jacobian(d.backend, d.f, d.xs) .* ys
+    end
+end
+
+function Base.:*(d::LazyJacobian, ys::AbstractArray)
+    if d.xs isa Tuple
+        return vec.(pushforward_function(d.backend, d.f, d.xs...)((ys,)))
+    else
+        vjp = (pushforward_function(d.backend, d.f, d.xs)((ys,)))
+        if vjp isa Tuple
+            return vec.(vjp)
+        else
+            return (vec(vjp),)
+        end
+    end
+end
+
+
 struct LazyHessian{B, F, X}
     backend::B
     f::F
     xs::X
 end
+
 function Base.:*(d::LazyHessian, ys)
-    return pushforward_function(
-        secondlowest(d.ab),
-        (xs...,) -> gradient(lowest(d.ab), d.f, xs...),
-        d.xs...,
-    )(ys)
+    if d.xs isa Tuple
+        return pushforward_function(
+            secondlowest(d.backend),
+                (xs...,) -> gradient(lowest(d.backend), d.f, xs...), d.xs...,)(ys)
+    else
+        return pushforward_function(
+            secondlowest(d.backend),
+                (xs,) -> gradient(lowest(d.backend), d.f, xs),d.xs,)(ys)
+    end
+
 end
+
 function Base.:*(ys, d::LazyHessian)
     if ys isa Tuple
         ya = adjoint.(ys)
     else
         ya = adjoint(ys)
     end
     return pullback_function(
-        secondlowest(d.ab),
-        (xs...,) -> gradient(lowest(d.ab), d.f, xs...),
+        secondlowest(d.backend),
+        (xs...,) -> gradient(lowest(d.backend), d.f, xs...),
         d.xs...,
     )(ya)
 end
 
+function Base.:*(d::LazyHessian, ys::Number)
+    if d.xs isa Tuple
+        return hessian(d.backend, d.f, d.xs...).*ys
+    else
+        return hessian(d.backend, d.f, d.xs).*ys
+    end
+end
+
+function Base.:*(ys::Number, d::LazyHessian)
+    if d.xs isa Tuple
+        return ys.*hessian(d.backend, d.f, d.xs...)
+    else
+        return ys.*hessian(d.backend, d.f, d.xs)
+    end
+end
+
+function Base.:*(d::LazyHessian, ys::AbstractArray)
+    if d.xs isa Tuple
+        return pushforward_function(
+            secondlowest(d.backend),
+                (xs...,) -> gradient(lowest(d.backend), d.f, xs...), d.xs...,)((ys,))
+    else
+        return pushforward_function(
+            secondlowest(d.backend),
+                (xs,) -> gradient(lowest(d.backend), d.f, xs),d.xs,)((ys,))
+    end
+end
+
+function Base.:*(ys::AbstractArray, d::LazyHessian)
```

Comment:
can we combine this with the general fallback?

Reply:
good idea! See my comment above as well -- When I started I thought it makes sense to dispatch them all separately because of these tiny differences between Arrays, Tuples, and Numbers as input but it really leads to much more code and gets less readable..